query/src/servers/http/middleware.rs

```diff
@@ -38,15 +38,22 @@ pub struct HTTPSessionMiddleware {
     pub session_manager: Arc<SessionManager>,
 }
 
-fn get_credential(headers: &HeaderMap) -> Result<Option<Credential>> {
-    let auth_headers: Vec<_> = headers.get_all(AUTHORIZATION).iter().collect();
+fn get_credential(req: &Request) -> Result<Credential> {
+    let auth_headers: Vec<_> = req.headers().get_all(AUTHORIZATION).iter().collect();
     if auth_headers.len() > 1 {
         let msg = &format!("Multiple {} headers detected", AUTHORIZATION);
         return Err(ErrorCode::AuthenticateFailure(msg));
     }
     if auth_headers.is_empty() {
-        return Ok(None);
+        return Err(ErrorCode::AuthenticateFailure(
+            "No authorization header detected",
+        ));
     }
+    let client_ip = match req.remote_addr().0 {
+        Addr::SocketAddr(addr) => Some(addr.ip().to_string()),
+        Addr::Custom(..) => Some("127.0.0.1".to_string()),
```

Comment:
where is this case used

Reply:
For testing,  `ep.call` will not have remote addr